src/cmd-buildextend-aws

```diff
@@ -102,9 +102,17 @@ def run_ore():
                 'snapshot': ore_data['SnapshotID']}
     buildmeta['amis'] = [ami_data]
 
-# Do it!
-generate_aws_vmdk()
+
+# Generate the disk image to be uploaded
+if 'aws' not in buildmeta['images']:
+    generate_aws_vmdk()
+else:
+    print('AWS image already built. Skipping image creation')
+# Do the upload if asked
 if args.upload:
     run_ore()
+else:
+    print('--upload not passed. Skipping image upload to cloud')
+# Update the build metadata
 write_json(buildmeta_path, buildmeta)
```

Comment:
Hmm, we should probably fold the `write_json` into the `generate_aws_vmdk()` and `run_ore()` functions. That way: (1) we don't touch the file if we didn't do anything, and (2) if ore is interrupted/errors, we still have e.g. the `aws` entry in the build metadata so that a rerun will skip the build.

Reply:
so you'd prefer we write the json twice (after each operation) rather than wait until the end?

Reply:
added in later commit